chia/full_node/block_store.py

```diff
@@ -123,34 +121,61 @@ def maybe_to_hex(self, field: bytes) -> Any:
 
     async def add_full_block(self, header_hash: bytes32, block: FullBlock, block_record: BlockRecord) -> None:
         self.block_cache.put(header_hash, block)
-        cursor_1 = await self.db.execute(
-            "INSERT OR REPLACE INTO full_blocks VALUES(?, ?, ?, ?, ?)",
-            (
-                self.maybe_to_hex(header_hash),
```

Comment:
The `maybe_to_hex` went away and I'm not sure why (or why it was there in the first place).

Reply:
`maybe_to_hex()` takes the db version into account and hex encodes the bytes in v1, but not in v2. It makes the code simple where we have a shared code path for both v1 and v2. However, this patch splits up the v1 and v2 path, since there are more fundamental changes (number of columns). Now, this line is only v1.

Reply:
Got it.

Reply:
This patch looks good. But regarding `.maybe_to_hex`, is there any disadvantage in always allowing either hex or binary to be read? That could allow for a lazy (maybe even *extremely* lazy) migration.

Reply:
I hadn't considered that. I'm not sure it would really help, since the v2 schema change goes deeper than just binary vs. hex.
I suppose it could be used as a middle-ground for a backwards-compatible v1 schema.